var/spack/repos/builtin/packages/dyninst/package.py

```diff
@@ -44,9 +45,19 @@ class Dyninst(Package):
     variant('stat_dysect', default=False,
             description="patch for STAT's DySectAPI")
 
+    # Dyninst depends on libelf and libdwarf prior to @9.3.0
+    # Dyninst depends on elfutils and libdwarf from @9.3.0 to but
+    # not including @develop
+    # Dyninst depends on elfutils and elfutils libdw from @develop forward
+    # elf@0 is an abstaction for libelf
+    # elf@1 is an abstaction for elfutils
     depends_on("elf@0", type='link', when='@:9.2.99')
+    # The sorting algorithm puts numbered releases as newer than alphabetic
+    # releases, but spack has special logic in place to ensure that
+    # develop is considered newer than all other releases.
+    # So, develop is included in the elf@1 line below.
     depends_on("elf@1", type='link', when='@9.3.0:')
-    depends_on("libdwarf")
+    depends_on("libdwarf", when='@:9.3.2')
```

Comment:
Are we certain that Dyninst will not depend on libdwarf if there is a 9.3.3 or 9.4 release? It could be the case that the libdwarf dependency is held until Dyninst 10. @jgalarowicz please confirm with the Dyninst folks at which version the libdwarf dependency will go away.

Reply:
It seems weird to me that a change like that would happen on a patch release (or a minor release, really).  I would just say `:9.3`.

Reply:
Hi Greg - It looks like the way it is coded is correct.

```
From: Sasha Da Rocha Pinheiro <darochapinhe@wisc.edu>
Yes, that is true.
Get Outlook for Android

From: Dyninst-api <dyninst-api-bounces@cs.wisc.edu> on behalf of Jim Galarowicz <jeg@krellinst.org>
Sent: Wednesday, June 13, 2018 11:21:30 AM
To: dyninst-api@cs.wisc.edu
Subject: [DynInst_API:] libdwarf usage in Dyninst?
Hi all,
I'm trying to get a spack file change to the dyninst/package.py file
through the spack review process.
My changes say that versions 9.3.2 and below use libdwarf and future
versions use libdw.
Is that true?
Thanks,
Jim G
```

Reply:
OK, I had the same thoughts as @tgamblin, but it's not my project...

Reply:
@tgamblin 9.3.2 was the last release using libdwarf.  I think the next release is going to be 10.0 and that will not use libdwarf.

Reply:
@jgalarowicz. That just muddies things more for me. If the need for libdwarf ends with Dyninst 10, then we should do as Todd suggests and do as Todd suggests with `:9`. IMHO we should do this in the Spack package even if Dyninst never plans another 9 release.

Reply:
Sorry, this really bothering me so I emailed Sasha back with my explicit concern:

```
Hi Greg,

I wanna say 9.3.2 is gonna be the last 9.x, since we are planing on releasing 10.x soon. But this is not set in stone. Could be a fix release in 9.x? It could, but very improbable. If it would be, it certainly would depend on libdwarf.

Sasha
________________________________________
From: Lee, Greg <lee218@llnl.gov>
Sent: Wednesday, June 13, 2018 1:03:55 PM
To: Sasha Da Rocha Pinheiro
Subject: FW: [spack/spack] Add the ability to build the dyninst master branch under the develop version name. (#8455) 
 
Hi Sacha,
 
I wanted to be a bit more explicit about Dyninst versions and dependence on libdwarf. I have 2 questions (the answer to the 1st may render the 2nd irrelevant) :
 
Is 9.3.2 going to be the last 9.X release ever?
 
If there ever is another 9.X release (i.e., 9.3.3 or 9.4), would it depend on libdwarf?
 
                -Greg
```

With this in mind, @jgalarowicz please make the `:9` changes (there are multiple lines that would need to be modified).

Reply:
@lee218llnl So, you want this line:
```
+    depends_on("libdwarf", when='@:9.3.2')
```
to be:
```
+    depends_on("libdwarf", when='@:9')
```
Will that still encompass the 9.3.2 and earlier versions?

Reply:
@lee218llnl Ok.  I see three places related to libdwarf and the @:9.3.2 versioning.  I will submit a change for that.